src/Components/test/Ignitor/CommentNode.cs

```diff
@@ -0,0 +1,15 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Ignitor
+{
+    internal class CommentNode : ContainerNode
```

Comment:
Is this called `CommentNode` because it plays the same role as the logical element containers in the TypeScript code? The TypeScript code happens to use DOM comment nodes as an implementation detail, but there's no reason it would have to be comment nodes if some other node type happened to be more suitable.

If the purpose of `CommentNode` here is to act as a logical element container, I suggest renaming it to `LogicalContainerNode` or something like that, rather than implying it needs to be a "comment".